src/client/eppo-client.ts

```diff
@@ -52,21 +52,31 @@ export default class EppoClient implements IEppoClient {
     validateNotBlank(experimentKey, 'Invalid argument: experimentKey cannot be blank');
     const experimentConfig = this.configurationStore.get<IExperimentConfiguration>(experimentKey);
     const allowListOverride = this.getSubjectVariationOverride(subjectKey, experimentConfig);
-    if (allowListOverride) {
-      return allowListOverride;
-    }
-    if (
-      !experimentConfig?.enabled ||
-      !this.subjectAttributesSatisfyRules(subjectAttributes, experimentConfig.rules) ||
-      !this.isInExperimentSample(subjectKey, experimentKey, experimentConfig)
-    ) {
+
+    if (allowListOverride) return allowListOverride;
+
+    // Check for disabled flags and configuration with no rules.
+    if (!experimentConfig?.enabled || experimentConfig.rules.length === 0) return null;
```

Comment:
I don't think the second condition is necessary; if the rules were empty, `matchesAnyRule` would return null, causing `getAssignment` to also return null.

Reply:
Good point.